ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -947,61 +525,50 @@ UIGestureRecognizerDelegate {
   /// - Parameters:
   ///   - isUpdate: Keep the keyboard files on failure
   public func downloadKeyboard(withID keyboardID: String, languageID: String, isUpdate: Bool) {
-    let kbInfo = [
-      Key.keyboardId: keyboardID,
-      Key.languageId: languageID
-    ]
-
-    guard downloadQueue == nil else {
-      // Download queue is active.
+    guard let keyboardsInfo = keyboardsInfo else {
+      let message = "Keyboard info has not yet been fetched. Call fetchKeyboardsList() first."
+      let error = NSError(domain: "Keyman", code: 0, userInfo: [NSLocalizedDescriptionKey: message])
+      downloadFailed(forKeyboards: [], error: error)
```

Comment:
There's a lot of code reordering in this section.  It looks to be correct, but it's hard for me to be sure.